platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/events/lifecycle/AppExitLogger.kt

```diff
@@ -114,30 +144,34 @@ internal class AppExitLogger(
         ) { APP_EXIT_EVENT_NAME }
     }
 
+    @WorkerThread
     fun logCrash(
         thread: Thread,
         throwable: Throwable,
     ) {
-        if (!runtime.isEnabled(RuntimeFeature.APP_EXIT_EVENTS)) {
-            return
-        }
-
-        // explicitly letting it run in the caller thread
-        logger.log(
-            LogType.LIFECYCLE,
-            LogLevel.ERROR,
-            buildCrashAndMemoryFieldsMap(thread, throwable),
-            blocking = true, // this ensures we block until the log has been persisted to disk
-        ) {
-            APP_EXIT_EVENT_NAME
-        }
+        runOnBackgroundThread {
```

Comment:
we should not make this call async... as you see below we want it to be blocking (`// this ensures we block until the log has been persisted to disk`), this is important because we will attempt to upload this log on next app launch so we should try to guarantee it will be indeed persisted